### PR TITLE
[Merged by Bors] - reduce hare commitee to 50

### PR DESCRIPTION
**Pull request by dshulyak**
this will reduce total gossipsub traffix x8.
comitee upgrade is scheduled to be used from layer 105_720 (July 15, 2024, 10:00:00 AM UTC).

### CODE REVIEW

**Comment by fasmat** [comment]
can we add this to `fastnet` config so we can check it in system tests?

**Comment by dshulyak** [comment]
it looks like i missed 2 places in oracle when i created this change.
if you plan to use it please update legacyOracle.go active and validate functions to use CommitteeFor helper

**Comment by poszu** [comment]
@dshulyak 
> it looks like i missed 2 places in oracle when i created this change. if you plan to use it please update legacyOracle.go active and validate functions to use CommitteeFor helper

What about the blocks/certifier.go? It currently uses a 200 committee. Can it be larger than Hare's committee size?

**Comment by poszu** [comment]
bors try

**Review by fasmat — APPROVED** [review]

File: systest/parameters/fastnet/smesher.json [hunk 1]
```diff
@@ -26,6 +26,12 @@
             }
         }
     },
+    "hare3": {
+        "committeeupgrade": {
+            "layer": 16,
+            "size": 50
+        }
+    },
```
Comment (dshulyak) [inline systest/parameters/fastnet/smesher.json:35]: this likely will be noop, as the number of nodes in tests are less then 50

**Comment by dshulyak** [comment]
> What about the blocks/certifier.go? It currently uses a 200 committee. Can it be larger than Hare's committee size?

yes, they are using same algo but otherwise independent.

where 50 came from? doesn't seem to make a difference, so just curious why you decided to change

**Comment by poszu** [comment]
> where 50 came from? doesn't seem to make a difference, so just curious why you decided to change

That's the number we agreed on in the team. There was no specific argument for 50 over 40 :shrug:.

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
Bors merge

**Comment by poszu** [comment]
bors merge

**Comment by fasmat** [comment]
The failing test indicates that the might be an issue with the change: [link](https://grafana.spacemesh.dev/explore?panes=%7B%22Qf4%22:%7B%22datasource%22:%22loki-gke-dev%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bnamespace%3D%5C%22test-pxfj%5C%22%7D%20%7C%3D%20%60WARN%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki-gke-dev%22%7D,%22editorMode%22:%22builder%22%7D%5D,%22range%22:%7B%22from%22:%221720258270725%22,%22to%22:%221720259571060%22%7D%7D%7D&schemaVersion=1&orgId=1)

```
2024-07-06 11:33:30.612 | {"L":"WARN","T":"2024-07-06T09:33:30.612Z","N":"node","M":"certificate committee size is not set, defaulting to hare committee size","size":800,"name":""}
```

shouldn't this be 50?

There are also errors by the `blockGenerator` and other components but I'm not sure what's the cause of the problems.

**Comment by dshulyak** [comment]
there was this failure in 17th layer. perhaps there is some unexpected interaction

```
2024-07-06 11:41:55.800	
{"L":"WARN","T":"2024-07-06T09:41:55.800Z","N":"node.hare","M":"failed","lid":17,"error":"hare failed to reach consensus in 2 iterations"}


	2024-07-06 11:42:43.566	
{"L":"INFO","T":"2024-07-06T09:42:43.566Z","N":"node.executor","M":"executed empty layer","sessionId":"3274113a","lid":17,"state_hash":"af1349b9f5f9a1a6a0404dea36dcc9499bcb25c9adc112b7cc9a93cae41f3262","duration":"191.612µs","name":"executor"}
	
	
2024-07-06 11:42:43.568	
{"L":"INFO","T":"2024-07-06T09:42:43.567Z","N":"node.executor","M":"executed block","sessionId":"3274113a","lid":18,"block":"48c4afd5b9","state_hash":"d20eb3e63b4be4cdb4b2d37fca0c290ee7b8169b2644cea40e8c97c70be73c1c","duration":"1.161017ms","count":0,"rewards":19,"name":"executor"}
	
	
2024-07-06 11:42:43.575	
{"L":"INFO","T":"2024-07-06T09:42:43.575Z","N":"node.executor","M":"executed block","sessionId":"3274113a","lid":19,"block":"d505834585","state_hash":"6b0728f7687f6349c6823fb1f607debab5ac2d8bd771585293379aa9fe2937f0","duration":"6.99174ms","count":30,"rewards":18,"name":"executor"}
	
	
2024-07-06 11:42:43.580	
{"L":"INFO","T":"2024-07-06T09:42:43.580Z","N":"node.executor","M":"executed block","sessionId":"3274113a","lid":20,"block":"7b6cf0a756","state_hash":"d27ff805a4937aa5f1fb6dd08ca02bbfc0f24881a277ea50a7bd32409dce211e","duration":"4.580587ms","count":12,"rewards":17,"name":"executor"}
	
	
2024-07-06 11:42:50.904	
{"L":"INFO","T":"2024-07-06T09:42:50.904Z","N":"node.executor","M":"executed block","lid":21,"block":"83da1e89e0","state_hash":"fa1a9acd73de7457206fc3b1a6fe6aa0d23e2a0051bea304ec9563b221bf65e8","duration":"2.161088ms","count":0,"rewards":20,"name":"executor"}
	
```

**Comment by dshulyak** [comment]
actually it looks accidental. pods were restarted in the middle of the test [messages rate](https://grafana.spacemesh.dev/explore?panes=%7B%22Sz7%22:%7B%22datasource%22:%22prometheus-gke-dev%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22sum%28rate%28spacemesh_p2p_processed_messages_duration_count%7Bkubernetes_namespace%3D%5C%22test-pxfj%5C%22,%20protocol%3D%5C%22%2Fh%2F3.0%5C%22%7D%5B5m%5D%29%29%20by%20%28kubernetes_pod_name%29%22,%22range%22:true,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22prometheus-gke-dev%22%7D,%22editorMode%22:%22code%22,%22legendFormat%22:%22__auto%22%7D%5D,%22range%22:%7B%22from%22:%221720258603583%22,%22to%22:%221720261119213%22%7D%7D%7D&schemaVersion=1&orgId=1)

image

**Comment by poszu** [comment]
> The failing test indicates that the might be an issue with the change: [link](https://grafana.spacemesh.dev/explore?panes=%7B%22Qf4%22:%7B%22datasource%22:%22loki-gke-dev%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bnamespace%3D%5C%22test-pxfj%5C%22%7D%20%7C%3D%20%60WARN%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki-gke-dev%22%7D,%22editorMode%22:%22builder%22%7D%5D,%22range%22:%7B%22from%22:%221720258270725%22,%22to%22:%221720259571060%22%7D%7D%7D&schemaVersion=1&orgId=1)
> 
> ```
> 2024-07-06 11:33:30.612 | {"L":"WARN","T":"2024-07-06T09:33:30.612Z","N":"node","M":"certificate committee size is not set, defaulting to hare committee size","size":800,"name":""}
> ```
> 
> shouldn't this be 50?

I was also puzzled by this. That's block certifier committee, not hare. It doesn't need to be equal to the hare committee.

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
bors merge

**Pull request closed without merging**